src/test/rgw/test_d4n_filter.cc

```diff
@@ -0,0 +1,2810 @@
+	     std::map<std::string, std::string>,
+	     std::map<std::string, std::string>,
+	     boost::redis::ignore_t > resp;
+
+    conn->async_exec(req, resp, yield[ec]);
+
+    ASSERT_EQ((bool)ec, false);
+    EXPECT_EQ((int)std::get<0>(resp).value(), 1);
+    EXPECT_EQ((int)std::get<1>(resp).value(), 1);
+    EXPECT_EQ(std::get<2>(resp).value().size(), 10);
+    EXPECT_EQ(std::get<3>(resp).value().size(), 10);
+
+    std::error_code err;
+    EXPECT_EQ(fs::exists(CACHE_DIR + "/" + TEST_BUCKET + testName + "/" + destName + "/dest_object_version", err), true);  
+
+    conn->cancel();
+    testBucket->remove(env->dpp, true, optional_yield{yield});
+    DriverDestructor driver_destructor(static_cast<rgw::sal::D4NFilterDriver*>(driver));
+  }, rethrow);
+
+  io.run();
+}
+
+TEST_F(D4NFilterFixture, DeleteObjectRead)
+{
+  const std::string testName = "DeleteObjectRead";
+ 
+  boost::asio::spawn(io, [this, &testName] (boost::asio::yield_context yield) {
+    init_driver(yield);
+    create_bucket(testName, yield);
+    put_object(testName, yield);
+
+    std::unique_ptr<rgw::sal::Object::ReadOp> read_op(obj->get_read_op());
+    ASSERT_EQ(read_op->prepare(optional_yield{yield}, env->dpp), 0);
+    ASSERT_EQ(read_op->iterate(env->dpp, 0, ofs, nullptr, optional_yield{yield}), 0);
+
+    std::string version;
+    {
+      boost::system::error_code ec;
+      request req;
+      req.push("EXISTS", TEST_BUCKET + testName + "_" + TEST_OBJ + testName + "_0_0");
+      req.push("EXISTS", TEST_BUCKET + testName + "__:null_" + TEST_OBJ + testName + "_0_0");
+      req.push("EXISTS", TEST_BUCKET + testName + "_" + TEST_OBJ + testName + "_0_9");
+      req.push("HGET", TEST_BUCKET + testName + "_" + TEST_OBJ + testName + "_0_0", "version"); 
+
+      response< int, int, int, std::string > resp;
+
+      conn->async_exec(req, resp, yield[ec]);
+      ASSERT_EQ((bool)ec, false);
+      EXPECT_EQ((int)std::get<0>(resp).value(), 1);
+      EXPECT_EQ((int)std::get<1>(resp).value(), 1);
+      EXPECT_EQ((int)std::get<2>(resp).value(), 1);
+      
+      version = std::get<3>(resp).value();
+    }
+
+    std::error_code err;
+    EXPECT_EQ(fs::exists(CACHE_DIR + "/" + TEST_BUCKET + testName + "/" + TEST_OBJ + testName + "/" + version, err), true);  
+    std::string oid = version + "#0#" + std::to_string(ofs);
+    EXPECT_EQ(fs::exists(CACHE_DIR + "/" + TEST_BUCKET + testName + "/" + TEST_OBJ + testName + "/" + oid, err), true);     
+
+    std::unique_ptr<rgw::sal::Object::DeleteOp> del_op = obj->get_delete_op();
+    EXPECT_EQ(del_op->delete_obj(env->dpp, optional_yield{yield}, rgw::sal::FLAG_LOG_OP), 0);
+
+    {
+      boost::system::error_code ec;
+      request req;
+      req.push("EXISTS", TEST_BUCKET + testName + "_" + TEST_OBJ + testName + "_0_0");
+      req.push("EXISTS", TEST_BUCKET + testName + "__:null_" + TEST_OBJ + testName + "_0_0");
+      req.push("EXISTS", TEST_BUCKET + testName + "_" + TEST_OBJ + testName + "_0_9");
+      req.push("FLUSHALL");
```

Comment:
what is the need for a FLUSHALL here?

Reply:
I add `FLUSHALL` calls at the end of each test when I am finished using redis in order to set up for the next test. I will simply move this to the setup method, however

Reply:
that looks better